src/main/groovy/edu/oregonstate/mist/locations/db/CampusMapDAO.groovy

```diff
@@ -39,10 +39,13 @@ class CampusMapDAO {
      */
     private List<CampusMapLocation> getCampusMapJson() {
         def jsonSlurper = new JsonSlurper()
-        String campusMapData = locationUtil.getDataFromUrlOrCache(
-                campusMapJsonUrl,
-                campusMapJsonOut)
-
-        jsonSlurper.parseText(campusMapData) as List<CampusMapLocation>
+        cache.withJsonFromUrlOrCache(campusMapJsonUrl, campusMapJsonOut) {
+            campusMapData ->
+                def locations = jsonSlurper.parseText(campusMapData) as List<CampusMapLocation>
+                if (locations.size() == 0) {
```

Comment:
You're doing a lot of checks for 0 or `isEmpty()` in the various dao classes. Should the cache be doing this check instead? Just trying to think of a way to abstract the condition of using a cached file or not, and it seems like the `Cache` class could be doing this.

Also, we probably only need to be checking for no locations in a datasource if we're using the non-cached version. The cached file should always be good data. I guess it doesn't hurt to check though.